repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.keystonev2
+
+import javax.servlet.http.HttpServletResponse.{SC_FORBIDDEN, SC_UNAUTHORIZED}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.filters.keystonev2.AbstractKeystoneV2Filter.Reject
+import org.openrepose.filters.keystonev2.KeystoneV2Common.{Endpoint, EndpointsData, Role, ValidToken}
+import org.openrepose.filters.keystonev2.config._
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+object KeystoneV2Authorization extends LazyLogging {
+
+  def doAuthorization(config: KeystoneV2Config, request: HttpServletRequestWrapper, validToken: ValidToken, endpoints: => Try[EndpointsData]): AuthorizationInfo = {
```

Comment:
I don't love the new case class that is basically a `Try` with more information on failure. That makes me think that this method should have its concerns separated somehow, but I don't think that is trivial to do. Furthermore, doing so would probably mean more code in both the authentication and authorization filters to do the same thing. At least, until we pull the authorization code out of the authentication filter. 

Perhaps this is just something to look at when we do finally yank the authorization code out of the authentication filter. After all, things become a bit simpler once we aren't worrying about scoping the token stuff that we write to headers.That is, the add everything then remove stuff the user isn't authorized to see approach that we get by default by separating the authentication and authorization filters seems cleaner and clearer than the current approach where we trying to figure out what to add before adding it to headers.

Reply:
Yeah, I wasn't super excited by it either, but due to how everything was currently wired together I didn't really see another way out, without massive rewrites, and even then i suspect we still would have ended with something similar.